tools/autograd/gen_variable_type.py

```diff
@@ -491,6 +494,11 @@ def format_trace_op_name(declaration):
 def format_trace_inputs(declaration):
     def dispatch_trace_input(arg_spec):
         name, value, simple_type, nullable = arg_spec
+        if declaration['use_c10_dispatcher'] == 'full':
+            if value == "options":
+                value = "TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory)"
+        else:
+            assert declaration['use_c10_dispatcher'] == 'with_codegenerated_unboxing_wrapper'
```

Comment:
Not in `tools/autograd` yet, alas.

Reply:
RIght, it's still raw data over here. But since we're now generating Declarations.yaml from groomed data, it seems reasonable to yank some of these (effectively no-op) sanity checks when it really helps to declutter the code. 

Not a huge deal either way, fine by me if you'd rather have them as an inducement to bring the data model over :)

Reply:
IDK, the assert is not making a statement about how clean or dirty the upstream source of the data is; it is there to make sure that *you* (the consumer) and upstream (the producer) agree on what the invariants are supposed to be, and complain loudly when they mismatch.